test/logflare/source/email_notification_server_test.exs

```diff
@@ -1,28 +1,22 @@
 defmodule Logflare.Source.EmailNotificationServerTest do
   @moduledoc false
   use Logflare.DataCase
+
   alias Logflare.Source.EmailNotificationServer
-  alias Logflare.Source.RecentLogsServer, as: RLS
-  alias Logflare.Sources
-  import Logflare.Factory
 
   setup do
     u1 = insert(:user)
-    s1 = insert(:source, user_id: u1.id)
-    sid = s1.token
-    rls = %RLS{source_id: sid, notifications_every: 1_000}
-    Sources.Counters.start_link()
-
-    {:ok, sources: [s1], args: rls}
+    s1 = insert(:source, user_id: u1.id, notifications_every: 1000)
+    [source: s1, user: u1]
   end
 
   describe "GenServer" do
-    test "start_link/1", %{sources: [_s1 | _], args: rls} do
-      assert {:ok, _pid} = EmailNotificationServer.start_link(rls)
+    test "start_link/1", %{source: source} do
+      start_supervised!({EmailNotificationServer, [source: source]})
```

Comment:
I like old version better, as it shows more explicitly what exactly we are testing there. At least `start_supervised` (non-bang) would read better IMHO.